### PR TITLE
GitHub enterprise support

### PR DESCRIPTION
**Pull request by marcellodesales**
Adds support for any Github Enterprise. Anyone with a host at `github.mycompany.com` can start another instance. This addresses Issue #141, allowing users to build the app for their GHE.
## Development
- Create a new application in the settings, choosing the icon for gitify http://iamemmanouil.com/static/images/projects/gitify.png
- Change the values of the protocol, host, api version and auth in `config.js`.
  - I just needed to change host `github.mycompany.com` and new values for the `clientId` and `clientSecret`.
- `npm run build`
- `npm start`

``` json
{
  "github": {
    "proto": "https",
    "host": "github.com",
    "apiVersion": "v3",
    "auth": {
      "clientId": "3fef4433a29c6ad8f22c",
      "clientSecret": "9670de733096c15322183ff17ed0fc8704050379"
    }
  }
}
```
## Possible Future Improvements
- Allow multiple accounts to be monitored, in addition to the public.
  - This avoids multiple apps to be opened in the bar.
  - The state of multiple apps overrides others when opened.
- Use a slack-like interface for allowing multiple instances to be monitored.
  - Allowing developers to see notifications from multiple instances is a win for me.

### CODE REVIEW

**Comment by gabhi** [comment]
:+1: Thank you.

**Comment by irae** [comment]
This feels like it will work with either enterprise or with public GitHub, but not both. It is supper common to need both. The config might need to become an array of accounts, and will probably need UI updates to config.

Suggestion: In the place I work, our GitHub enterprise uses a custom 2-legged web-auth, almost no clients support it correctly. The successful tools use manually generated tokens to setup configuration. I use "hub" the command line tool and it works by writing a token to their dotfile.

File: src/js/utils/config-manager.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  hostUrl: function getHostUrl() {
+    return config.github.proto + '://' + config.github.host;
+  },
+  apiUrl: function getApiUrl() {
+    return config.github.host.indexOf('github.com') === -1 ?
+      this.hostUrl() + '/api/' + config.github.apiVersion :
+      config.github.proto + '://api.' + config.github.host;
+  },
+  apiHost: function getApiHostOnly() {
+    return this.apiUrl().replace('https://', '').replace('http://', '');
```
Comment (Oliboy50) [inline src/js/utils/config-manager.js]: `return this.apiUrl().replace(config.github.proto + '://'', '');`

**Comment by Oliboy50** [comment]
should this PR be merged or closed?
Gitify still only supports Github (not Enterprise)?

**Comment by ekonstantinidis** [comment]
Hi there, so there good and bad news. The app has been through a complete rewrite (good news - v1.0.0) so this PR unfortunately won't work. The idea is 100% correct but we have to rewrite. 

I do not have access to an enterprise account (the GitHub team provided me with a license for development) but I don't have a server that is able to handle(hardware-wise) GitHub Enterprise. 

If someone is able to provide me with an account then I will be happy to spend some time on implementing GitHub Enterprise support.

**Comment by nabeelio** [comment]
@ekonstantinidis what's required, server-wise? I can spinup something on Digital Ocean for you to use.

**Comment by diranged** [comment]
Any news on this PR?

**Comment by manosim** [comment]
Hello everyone! So @deibeljc had some amazing progress on this one which is based on the `v2` branch. Hopefully we will be able to release this one soon (I'm currently on holidays so I've finally have the time to do some OS work!). I'm closing this one and we can follow up on #279.

Ps. Let me know if there is any chance you can provide me with a GitHub Enterprise account so I can do some development/tests (just a dummy/empty repository to generate some notifications would be great - thank you!).

**Pull request closed without merging**